src/js/views/Projects/Projects.jsx

```diff
@@ -1,57 +1,157 @@
 import React, { useContext, useEffect, useState } from 'react'
-import { Link, useNavigate, useLocation } from 'react-router-dom'
 import { useTranslation } from 'react-i18next'
+import {
+  fromKueryExpression,
+  toElasticsearchQuery
+} from '@cybernetex/kbn-es-query'
+import { byString, byNumber, byValues } from 'sort-es'
 
-import { Alert, Icon, Loading } from '../../components'
+import { Alert, ScoreBadge, Table } from '../../components'
 import { Context } from '../../state'
-import { httpGet } from '../../utils'
+import { httpPost } from '../../utils'
 
-import { DataTable } from './DataTable'
 import { Filter } from './Filter'
 
 import { metadataAsOptions } from '../../settings'
 
-function buildSortDefault(sort) {
-  const value = {}
-  const sortMatches = sort.match(
-    /(?:(name|namespace|project_score|project_type) (asc|desc))/g
-  )
-  if (sortMatches !== null) {
-    sortMatches.map((match) => {
-      const [column, direction] = match.split(' ')
-      value[column] = direction
-    })
-  }
-  if (Object.keys(value).length === 0) {
-    value.namespace = 'asc'
-    value.name = 'asc'
-  }
+const sortMap = {
+  namespace: byString,
+  name: byString,
+  project_score: byNumber,
+  type: byString
+}
+
+function slugToName(items, slug) {
+  let value = slug
+  items.forEach((item) => {
+    if (item.slug === slug) {
+      value = item.name
+    }
+  })
   return value
 }
 
+function sortTableData(data, columns) {
+  const sortSettings = []
+  for (const [column, direction] of Object.entries(columns)) {
+    sortSettings.push([column, sortMap[column]({ desc: direction === 'desc' })])
+  }
+  return data.sort(byValues(sortSettings))
+}
+
 function Projects() {
   const [errorMessage, setErrorMessage] = useState(null)
   const [globalState, dispatch] = useContext(Context)
-  const navigate = useNavigate()
-  const query = new URLSearchParams(useLocation().search)
-  const sortOrder = ['namespace', 'project_type', 'name', 'project_score']
+  const { t } = useTranslation()
+
+  function buildURL(path) {
+    return new URL(path, globalState.baseURL)
+  }
+
+  function onFilterChange(value) {
+    if (value !== globalState.projects.filter) {
+      setState({ ...state, data: [] })
+      dispatch({
+        type: 'SET_PROJECTS_FILTER',
+        payload: value
+      })
+    }
+  }
+
+  function onRefresh() {
+    setState({ ...state, data: [], refresh: true })
+  }
+
+  function onSortChange(column, direction) {
+    const sort = { ...globalState.projects.sort }
+    if (direction === null) {
+      if (sort[column] !== undefined) delete sort[column]
+    } else if (globalState.projects.sort[column] !== direction) {
+      sort[column] = direction
+    }
+    if (globalState.projects.sort !== sort) {
```

Comment:
This will always be true because `sort` is a new object and JS doesn't do structural equality by default.

Reply:
I was actually wondering about that... _add to reasons that I do not consider myself a JS programmer_.

Reply:
Same, addressed.